lib/markdown_videos/renderer.rb

```diff
@@ -13,42 +13,26 @@ def initialize(string, options = {})
     end
 
     def render
-      MarkdownVideos::SERVICES.keys.each do |service|
-        @string = render_service(service)
-      end
-      @string
-    end
+      @string.gsub(/!\[([^\]]*)\]\(([^)]+)\)/) do
+        rendered = nil
+        match_data = Regexp.last_match
```

Comment:
Argument given to the block only [returns the `string` part of `match_data`](http://brettterpstra.com/2014/10/28/the-matchdata-object-in-ruby-gsub-blocks/), strangely enough. It's why you need to use `Regexp.last_match`.